docs/source/user-guide/dataset_object.rst

```diff
@@ -0,0 +1,128 @@
+.. _dataset_object:
+
+====================
+The Dataset Object
+====================
+The dataset is one of the basic blocks of deepchecks. It is a container for the data and it's relevant metadata, like special column names (label, date, index, etc).
+Some of the checks allows to use a dataframe directly, but some others requires the metadata in order to run, so they are limited to working only with Datasets.
+
+Class Parameters
+===================
+All of the parameters are optional.
+
+.. list-table::
+    :widths: 20 20 50 10
+    :header-rows: 1
+
+    * - Name
```

Comment:
Just thinking about how can we maintain this when we change the dataset parameters/signature..

Reply:
Yes I agree that this isn't sustainable over time. It can be useful for now (assuming it's still updated)

I discussed this with @matanper, who is the one that wrote this specific page :) (I just appended it as part of the PR because the context was relevant and he was away) that we should shorten this description, and add a link to the API reference... However for now let's keep it as is and work on it as a separate small task (because till we update it I think it's still better to have this version rather than not have anything)

Reply:
#366